Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/corefinement.h

```diff
@@ -38,14 +39,94 @@ namespace Polygon_mesh_processing {
 namespace Corefinement
 {
 /** \ingroup PMP_corefinement_grp
- *  %Default new-face visitor model of `PMPCorefinementVisitor`.
+ *  %Default visitor model of `PMPCorefinementVisitor`.
  *  All of its functions have an empty body. This class can be used as a
  *  base class if only some of the functions of the concept require to be
  *  overridden.
  */
 template <class TriangleMesh>
 struct Default_visitor;
 
+#ifdef DOXYGEN_RUNNING
+/** \ingroup PMP_corefinement_grp
+ * A model of `PMPCorefinementVisitor` that allows to extract non-manifold
+ * outputs from a corefinement based Boolean Operations.
+ *
+ * \tparam TriangleMesh a model of `HalfedgeListGraph`, `FaceListGraph`, and `MutableFaceGraph`
+ * \tparam VPM1 a class model of `ReadablePropertyMap` with `boost::graph_traits<TriangleMesh>::%vertex_descriptor`
+ *              as key type and an unspecified type `%Point_3` as value type
+ * \tparam VPM2 a class model of `ReadablePropertyMap` with `boost::graph_traits<TriangleMesh>::%vertex_descriptor`
+ *              as key type and an unspecified type `%Point_3` as value type
+ */
+template <class TriangleMesh,
+          class VPM1 = typename boost::property_map<TriangleMesh, vertex_point_t>::type,
+          class VPM2 = typename boost::property_map<TriangleMesh, vertex_point_t>::type>
+struct Visitor_for_non_manifold_output
+{
+/**
+ * \param tm1 first triangle mesh in the same order as in the function in \ref PMP_corefinement_grp
+ * \param tm2 second triangle mesh in the same order as in the function in \ref PMP_corefinement_grp
+ * \param vpm1 vertex point map for `tm1`
+ * \param vpm2 vertex point map for `tm2`
+ */
+  Visitor_for_non_manifold_output(TriangleMesh& tm1,
```

Comment:
Does the visitor keep references to the meshes? Might wanna make explicit that it does and that it should be precisely the same meshes, and not a copy or whatnot.

These functions do not have the options to not modify input meshes right?

Can the visitor be re-used if you re-use the meshes?